cardinal/server/server.go

```diff
@@ -46,6 +53,152 @@ const (
+	specDoc, err := loads.Spec(pathToSwaggerSpec)
+	if err != nil {
+		return nil, err
+	}
+	api := untyped.NewAPI(specDoc)
+	err = registerTxHandlerSwagger(w, api)
+	if err != nil {
+		return nil, err
+	}
+	err = registerReadHandlerSwagger(w, api)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := api.Validate(); err != nil {
+		log.Fatalln(err)
+	}
+
+	app := middleware.NewContext(specDoc, api, nil)
+	th := &Handler{
+		w:   w,
+		mux: http.NewServeMux(),
+	}
+	for _, opt := range opts {
+		opt(th)
+	}
+	th.mux.Handle("/", app.APIHandler(nil))
+	th.initialize()
+	return th, nil
+}
+
+// utility function to extract parameters from swagger handlers
+func getValueFromParams(params interface{}, name string) (interface{}, bool) {
+	data, ok := (params).(map[string]interface{})
+	if !ok {
+		return nil, ok
+	}
+	return data[name], true
+}
+
+// register transaction handlers on swagger server
+func registerTxHandlerSwagger(world *ecs.World, api *untyped.API) error {
+	//var txEndpoints []string
+
+	// Register tx endpoints
+	txs, err := world.ListTransactions()
+	if err != nil {
+		return err
+	}
+
+	txNameToTx := make(map[string]transaction.ITransaction)
+	for _, tx := range txs {
+		txNameToTx[tx.Name()] = tx
+	}
+
+	coreHandler := runtime.OperationHandlerFunc(func(params interface{}) (interface{}, error) {
+		return TransactionReply{
+			TxHash: "",
+			Tick:   0,
+		}, errors.New("not implemented")
+	})
+
+	createPersonaHandler := runtime.OperationHandlerFunc(func(params interface{}) (interface{}, error) {
+		return ecs.CreatePersonaTransactionResult{Success: true}, errors.New("not implemented")
+	})
+
+	authorizePersonaAddressHandler := runtime.OperationHandlerFunc(func(i interface{}) (interface{}, error) {
+		return ecs.AuthorizePersonaAddressResult{Success: true}, errors.New("not implemented")
+	})
+	api.RegisterOperation("POST", "/tx/core/{txType}", coreHandler)
+	api.RegisterOperation("POST", "/tx/persona/create-persona", createPersonaHandler)
+	api.RegisterOperation("POST", "/tx/persona/authorize-persona-address", authorizePersonaAddressHandler)
+
+	return nil
+}
+
+// result struct for /query/http/endpoints
+type EndpointsResult struct {
+	TxEndpoints    []string `json:"tx_endpoints"`
+	QueryEndpoints []string `json:"query_endpoints"`
+}
+
+// register query endpoints for swagger server
+func registerReadHandlerSwagger(world *ecs.World, api *untyped.API) error {
+	//var txEndpoints []string
+	coreHandler := runtime.OperationHandlerFunc(func(params interface{}) (interface{}, error) {
+		return struct {
+			test string
+		}{test: "test"}, errors.New("not implemented")
+	})
+	listHandler := runtime.OperationHandlerFunc(func(params interface{}) (interface{}, error) {
+		txs, err := world.ListTransactions()
+		txEndpoints := make([]string, 0, len(txs))
+		if err != nil {
+			return nil, err
+		}
+		for _, tx := range txs {
+			if tx.Name() != "create-persona" && tx.Name() != "authorize-persona-address" {
```

Comment:
This line should be replaced with:

`if tx.Name() != ecs.CreatePersonaTx.Name() && tx.Name() != ecs.AuthorizePersonaAddressTx.Name() {...`